src/global_scheduler/global_scheduler.cc

```diff
@@ -14,6 +14,45 @@
 #include "state/task_table.h"
 
 /**
+ * Retry the task assignment. If the local scheduler that the task is assigned
+ * to is no longer active, do not retry the assignment.
+ * TODO(rkn): We currently only retry the method if the global scheduler
+ * publishes a task to a local scheduler before the local scheduler has
+ * subscribed to the channel. If we enforce that ordering, we can remove this
+ * retry method.
+ *
+ * @param id The task ID.
+ * @param user_context The global scheduler state.
+ * @param user_data The Task that failed to be assigned.
+ * @return Void.
+ */
+void assign_task_to_local_scheduler_retry(UniqueID id,
+                                          void *user_context,
+                                          void *user_data) {
+  GlobalSchedulerState *state = (GlobalSchedulerState *) user_context;
+  Task *task = (Task *) user_data;
+  CHECK(Task_state(task) == TASK_STATUS_SCHEDULED);
+
+  // If the local scheduler has died since we requested the task assignment, do
+  // not retry again.
+  DBClientID local_scheduler_id = Task_local_scheduler(task);
+  auto it = state->local_schedulers.find(local_scheduler_id);
+  if (it == state->local_schedulers.end()) {
```

Comment:
In the case where the local scheduler has died, don't we need to do something? E.g., mark it as waiting again or call `handle_task_waiting` on it? Otherwise we could lose the task, right?

Reply:
Nope, the monitor will mark the task as lost eventually. It would be better not to write to the task table again, in case we interfere with someone trying to reconstruct the task.